ignite/metrics/rouge.py

```diff
@@ -0,0 +1,437 @@
+    return Score(match=match, candidate=len(candidate), reference=len(reference))
+
+
+class MultiRefReducer(metaclass=ABCMeta):
+    r"""
+    Reducer interface for multi-reference
+    """
+
+    @abstractmethod
+    def __call__(self, scores: Sequence[Score]) -> Score:
+        pass
+
+
+class MultiRefAverageReducer(MultiRefReducer):
+    r"""
+    Reducer for averaging the scores
+    """
+
+    def __call__(self, scores: Sequence[Score]) -> Score:
+        match = sum([score.match for score in scores])
+        candidate = sum([score.candidate for score in scores])
+        reference = sum([score.reference for score in scores])
+        return Score(match=match, candidate=candidate, reference=reference)
+
+
+class MultiRefBestReducer(MultiRefReducer):
+    r"""
+    Reducer for selecting the best score
+    """
+
+    def __call__(self, scores: Sequence[Score]) -> Score:
+        return max(scores, key=lambda x: x.recall)
+
+
+class _BaseRouge(Metric):
+    r"""
+    Rouge interface for Rouge-L and Rouge-N
+    """
+
+    def __init__(
+        self,
+        multiref: str = "average",
+        alpha: float = 0,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+        super(_BaseRouge, self).__init__(output_transform=output_transform, device=device)
+        self._alpha = alpha
+        if not 0 <= self._alpha <= 1:
+            raise ValueError(f"alpha must be in interval [0, 1] (got : {self._alpha})")
+        self._multiref = multiref
+        valid_multiref = ["best", "average"]
+        if self._multiref not in valid_multiref:
+            raise ValueError(f"aggregation must be in {valid_multiref} (got : {self._multiref })")
+        self._mutliref_reducer = self._get_multiref_reducer()
+
+    def _get_multiref_reducer(self) -> MultiRefReducer:
+        if self._multiref == "average":
+            return MultiRefAverageReducer()
+        if self._multiref == "best":
+            return MultiRefBestReducer()
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._recall = 0
+        self._precision = 0
+        self._fmeasure = 0
+        self._num_examples = 0
+
+    @reinit__is_reduced
+    def update(self, output: Tuple[Sequence[Any], Sequence[Sequence[Any]]]) -> None:
+        candidate, references = output[0], output[1]
+        multiref_scores = [self.compute_score(candidate=candidate, reference=reference,) for reference in references]
+        score = self._mutliref_reducer(multiref_scores)
+        self._precision += score.precision
+        self._recall += score.recall
+        precision_recall = score.precision * score.recall
+        if precision_recall > 0:  # avoid zero division
+            self._fmeasure += precision_recall / ((1 - self._alpha) * score.precision + self._alpha * score.recall)
+        print("-********", candidate, references)
+        print(score.precision, score.recall, self._fmeasure)
```

Comment:
Do we need to keep this prints ?

Reply:
Sorry, I would have removed such debug prints...